backend/pkg/helm/release.go

```diff
@@ -0,0 +1,744 @@
+package helm
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"errors"
+	"net/http"
+
+	"github.com/go-playground/validator/v10"
+	"github.com/gorilla/schema"
+
+	"github.com/rs/zerolog"
+	zlog "github.com/rs/zerolog/log"
+	"gopkg.in/yaml.v2"
+	"helm.sh/helm/v3/pkg/action"
+	"helm.sh/helm/v3/pkg/chart"
+	"helm.sh/helm/v3/pkg/chart/loader"
+	"helm.sh/helm/v3/pkg/downloader"
+	"helm.sh/helm/v3/pkg/getter"
+	"helm.sh/helm/v3/pkg/release"
+	"helm.sh/helm/v3/pkg/storage/driver"
+)
+
+type ListReleaseRequest struct {
+	AllNamespaces *bool   `json:"allNamespaces,omitempty"`
+	Namespace     *string `json:"namespace,omitempty"`
+	All           *bool   `json:"all,omitempty"`
+	ByDate        *bool   `json:"byDate,omitempty"`
+	Limit         *int    `json:"limit,omitempty"`
+	Offset        *int    `json:"offset,omitempty"`
+	Filter        *string `json:"filter,omitempty"`
+	Uninstalled   *bool   `json:"uninstalled,omitempty"`
+	Superseded    *bool   `json:"superseded,omitempty"`
+	Uninstalling  *bool   `json:"uninstalling,omitempty"`
+	Deployed      *bool   `json:"deployed,omitempty"`
+	Failed        *bool   `json:"failed,omitempty"`
+	Pending       *bool   `json:"pending,omitempty"`
+}
+
+type ListReleaseResponse struct {
+	Releases []*release.Release `json:"releases"`
+}
+
+// Returns (releases, error) given the request and helm configuration.
+//
+//nolint:gocognit
+func getReleases(req ListReleaseRequest, config *action.Configuration) ([]*release.Release, error) {
+	// Get list client
+	listClient := action.NewList(config)
+
+	// Removing all these if assignments is not possible, so we disable gocognit linter
```

Comment:
I'm not sure if all of these assignments can be done in a cleaner way? 

Is there a single line way to do this? I know there's no ternary operators... but something like this? (untested)
`listClient.AllNamespaces = req.AllNamespaces != nil && *req.AllNamespaces || nil`

Reply:
We can create a separate function to initialize the listClient from the request params as a work around for the gocognit linter.

Reply:
This code cleanup could happen in future work too IMHO. Resolving this convo.